### ISSUE TITLE
Failure Example States Sink Subscriber Ignoring Completions 

### ISSUE THREAD

**Issue by MaxMacleod** [mattneub/understandingCombine#19]
Hi,
in this [page](https://www.apeth.com/UnderstandingCombine/start/startfailures.html) the text states:

> Currently, our sink subscriber is ignoring completions, including failures; that is the significance of the empty receiveCompletion function! 

but the example has code in the completion to assign the output:

`self.iv.image = $0`

So strictly speaking the sink subscriber is not ignoring the completion.

Hope this helps!

Best Wishes,
Max

**Comment by mattneub** [comment]
No, I think you've misunderstood the text here. `.sink` takes _two_ handlers, one for completions, one for values. The example code says:

    .sink(receiveCompletion: {_ in}) {
        self.iv.image = $0
    }

The `self.iv.image = $0` is, as you say, assigning the _output_, meaning any _value_ that comes down the pipeline. But any _completions_ that come down the pipeline are routed into the `receiveCompletion` handler, which is

    {_ in}

That does indeed ignore its parameter entirely; it is, as the text says, empty. If a completion arrives, the `.sink`, as we have it written, will do nothing. And that includes a failure.
